src/tools/singlejar/output_jar_shell_test.sh

```diff
@@ -13,22 +13,44 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#
-#
-# Arguments:
-#   unittest.bash script
-#   singlejar path
-#   jar tool path
 
-(($# >= 2)) || \
-  { echo "Usage: $0 <unittest.bash dir> <singlejar>" >&2; exit 1; }
+set -euo pipefail
+# --- begin runfiles.bash initialization ---
+if [[ ! -d "${RUNFILES_DIR:-/dev/null}" && ! -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+    if [[ -f "$0.runfiles_manifest" ]]; then
+      export RUNFILES_MANIFEST_FILE="$0.runfiles_manifest"
+    elif [[ -f "$0.runfiles/MANIFEST" ]]; then
+      export RUNFILES_MANIFEST_FILE="$0.runfiles/MANIFEST"
+    elif [[ -f "$0.runfiles/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+      export RUNFILES_DIR="$0.runfiles"
+    fi
+fi
+if [[ -f "${RUNFILES_DIR:-/dev/null}/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+  source "${RUNFILES_DIR}/bazel_tools/tools/bash/runfiles/runfiles.bash"
+elif [[ -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+  source "$(grep -m1 "^bazel_tools/tools/bash/runfiles/runfiles.bash " \
+            "$RUNFILES_MANIFEST_FILE" | cut -d ' ' -f 2-)"
+else
+  echo >&2 "ERROR: cannot find @bazel_tools//tools/bash/runfiles:runfiles.bash"
+  exit 1
+fi
+# --- end runfiles.bash initialization ---
 
 # Load test environment
-source $1/unittest.bash \
+source "$(rlocation "io_bazel/src/test/shell/unittest.bash")" \
   || { echo "unittest.bash not found!" >&2; exit 1; }
 
-set -e
-declare -r singlejar="$2"
+IS_WINDOWS=false
+case "$(uname | tr [:upper:] [:lower:])" in
+msys*|mingw*|cygwin*)
+  IS_WINDOWS=true
```

Comment:
opt: You could set `EXE_EXT=".exe"` for Windows and and "" for Linux, and then do away with the `if-then-else` in line 49 and have just `singlejar="$(rlocation "io_bazel/src/tools/singlejar/singlejar${EXE_EXT}")"`.

Also in the other test.

Reply:
Done.